packages/executor/src/execution/promiseWithResolvers.ts

```diff
@@ -0,0 +1,20 @@
+import { MaybePromise } from '@graphql-tools/utils';
+
+/**
+ * Based on Promise.withResolvers proposal
+ * https://github.com/tc39/proposal-promise-with-resolvers
+ */
+export function promiseWithResolvers<T>(): {
```

Comment:
I think this is extra. We already have `createDeferred`, no?

Reply:
sounds good

Reply:
actually, then createDeferred should probably be moved to utils, right?

Reply:
Yes makes sense!

Reply:
done in 1ce4b1ef